src/main/java/org/broadinstitute/hellbender/utils/variant/GATKVariantContextUtils.java

```diff
@@ -1137,20 +1224,121 @@ private static boolean likelihoodsAreUninformative(final double[] likelihoods) {
      * @param allelesToKeep   the list of alleles to keep
      * @return non-null bitset
      */
-    private static boolean[] getAlleleIndexBitset(final VariantContext originalVC, final List<Allele> allelesToKeep) {
+    private static BitSet getAlleleIndexBitset(final VariantContext originalVC, final List<Allele> allelesToKeep) {
         final int numOriginalAltAlleles = originalVC.getNAlleles() - 1;
-        final boolean[] alleleIndexesToKeep = new boolean[numOriginalAltAlleles + 1];
+        final BitSet alleleIndexesToKeep = new BitSet(numOriginalAltAlleles + 1);
 
         // the reference Allele is definitely still used
-        alleleIndexesToKeep[0] = true;
+        alleleIndexesToKeep.set(0);
         for ( int i = 0; i < numOriginalAltAlleles; i++ ) {
             if ( allelesToKeep.contains(originalVC.getAlternateAllele(i)) )
-                alleleIndexesToKeep[i+1] = true;
+                alleleIndexesToKeep.set(i+1);
         }
 
         return alleleIndexesToKeep;
     }
 
+    /**
+     * Find the strand allele count indexes to use for a selected set of alleles
+     *
+     * @param originalVC   the original VariantContext
+     * @param allelesToUse the subset of alleles to use
+     * @return a list of SAC indexes to use or null if none
+     */
+    public static List<Integer> determineSACIndexesToUse(final VariantContext originalVC, final List<Allele> allelesToUse) {
+
+        Utils.nonNull(originalVC, "VariantContext is null");
+        Utils.nonNull(allelesToUse, "Alleles to use is null");
+
+        // the bitset representing the allele indexes we want to keep
+        final BitSet alleleIndexesToUse = getAlleleIndexBitset(originalVC, allelesToUse);
+
+        // an optimization: if we are supposed to use all (or none in the case of a ref call) of the alleles,
+        // then we can keep the SACs as is; otherwise, we determine which ones to keep
+        if (alleleIndexesToUse.cardinality() == alleleIndexesToUse.size()) {
+            return null;
+        }
+
+        return getSACIndexes(alleleIndexesToUse);
+    }
+
+    /**
+     * Get the actual strand aleele counts indexes to use given the corresponding allele indexes
+     *
+     * @param alleleIndexesToUse    the bitset representing the alleles to use (@see #getAlleleIndexBitset)
+     * @return a non-null List
+     */
+    private static List<Integer> getSACIndexes(final BitSet alleleIndexesToUse) {
+
+        Utils.nonNull(alleleIndexesToUse, "Alleles to use is null");
+        if (alleleIndexesToUse.isEmpty()) throw new IllegalArgumentException("cannot have no alleles to use");
+
+        final List<Integer> result = new ArrayList<>(2 * alleleIndexesToUse.size());
+
+        for (int SACindex = 0; SACindex < alleleIndexesToUse.size(); SACindex++) {
+            if (alleleIndexesToUse.get(SACindex)) {
+                result.add(2 * SACindex);
```

Comment:
It's the number of read counts per allele per strand. The 2 is the number of strand (forward and reverse).
https://software.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_gatk_tools_walkers_annotator_StrandAlleleCountsBySample.php
I will add a comment.